src/lib/environment/crateinfo.rs

```diff
@@ -185,3 +187,41 @@ pub(crate) fn load_from(file_path: PathBuf) -> CrateInfo {
         CrateInfo::new()
     }
 }
+
+pub(crate) fn crate_target_triple(path: impl Into<PathBuf>) -> Option<String> {
+    let path = path.into();
+    let mut target_triple = None;
+
+    let config_folders = path
+        .ancestors()
+        .map(|ancestor| ancestor.join(".cargo"))
+        .chain(env::var("CARGO_MAKE_CARGO_HOME").map(PathBuf::from));
```

Comment:
what happens if CARGO_MAKE_CARGO_HOME is not defined?

Reply:
Nothing.
```rust
.chain({
    let home: Option<String> = env::var("CARGO_MAKE_CARGO_HOME").map(PathBuf::from);
    home.into_iter()
});
```
`chain` takes an iterator, if `Option` is `None` it will just be an empty iterator.